src/sas/qtgui/Perspectives/Fitting/FittingUtilities.py

```diff
@@ -679,82 +677,85 @@ def getWeight(data, is2d, flag=None):
+                 being checked or not
+    :type flag: bool
     :return: Weight multiplier for each dataset
     :rtype: dict
     """
 
     stat_weights = {}
-    subject_stat_weights = {}
     weight_increase = {}
-    num_fits = len(weights.keys())
 
-    # If weighting = None in Fit Options tab
-    if flag == 0:
-        for id_index in weights.keys():
-            weight_increase[id_index]
-        return weight_increase
+    # If "Modify weighting" option not checked
+    if not flag:
+        return {k: 1.0 for k in weights}
 
-    # Calc statistical weight for each dataset
+    # Calc statistical weight for each dataset and maximum
+    # Need to find the best "weighting" scheme
+    max_weight = 0
     for id_index in weights.keys():
-        stat_weight = 0.0
-        for val in weights[id_index]:
-            stat_weight += 1.0 / (val ** 2.0)
-        av_stat_weight = stat_weight / len(weights[id_index])
-        stat_weights[id_index] = av_stat_weight
-        if ratios[id_index] == "Subject":
-            subject_stat_weights[id_index] = av_stat_weight
-            weight_increase[id_index] = 1.0
-        elif ratios[id_index] == "Default":
-            weight_increase[id_index] = 1.0
-
-    # If all data sets defined, i.e. are "Default" then no need to continue
-    if len(weight_increase.keys()) == len(weights.keys()):
-        return weight_increase
-
-    # If no data set defined as subject, use the average statistical weight as the comparison point.
-    if len(subject_stat_weights.keys()) == 0:
-        subject_stat_weight = sum([v for v in stat_weights.values()]) / num_fits
-    # If one data set defined as subject
-    elif len(subject_stat_weights.keys()) == 1:
-        subject_stat_weight = list(subject_stat_weights.values())[0]
-    # If multiple data set defined as subject, then use first
-    elif len(subject_stat_weights.keys()) > 1:
-        first_subject_stat_weight_id = list(subject_stat_weights.keys())[0]
-        logging.warning(f'Multiple data sets specified for comparison using "Subject", will use '
-                        f'{first_subject_stat_weight_id} for comparison.')
-        subject_stat_weight = subject_stat_weights[first_subject_stat_weight_id]
-
-    for id_index, weight in stat_weights.items():
-        # If dataset is fixed or compare, don't modify
-        if id_index in weight_increase.keys():
-            continue
-        else:
-            desired_stat_weight = subject_stat_weight * float(ratios[id_index])
-            difference = desired_stat_weight / weight
-            weight_increase[id_index] = math.sqrt(difference) / num_fits
+        stat_weights[id_index] = numpy.sum(1.0/weights[id_index]**2)
+        if stat_weights[id_index] > max_weight:
+            max_weight = stat_weights[id_index]
+
+    for id_index in weights.keys():
+            weight_increase[id_index] = float(ratios[id_index]) * max_weight / stat_weights[id_index]
```

Comment:
Thanks for the comments. I agree with all of them, but for the moment I only implemented the change to use min(w) instead of max(w). This should not affect the parameters, but it certainly gives more credible uncertainties.

I agree that it is not a good idea to play with the weights, and there is a clear warning about the dangers of using this option. It is completely empirical and I don't know any math/stats reference to support it. I'm writing a SasView tuto where I plan to explain in more detail the dangers and I will likely include a large part of the discussion that you had with Iestyn. 

I also agree that the weights to be sent to bumps should be square rooted. Again I decided not to do it based only on empirical reasons. In a test using 2 data sets I found that I could get an intermediate answer using user-weights of 1 and 3, while the needed user-weights were 500 and 1 when returning the sqrt. But this was a single test and perhaps not very representative, so further tests should be done and this could be changed.

Reply:
While testing, I found that all the FitPages shown in the simultaneous fit tab are fitted, even if a page is not checked to be fit. Furthermore, any change in the tab (e.g. adding a constraint, modifying a weight, etc.) reinitializes the status of the checkbuttons. And the weights are applied to all the FitPages, so I probably need to include again a "Default" option.

Reply:
Some possibilities: 
1. You have an old version of bumps that doesn't have the w² scaling, but only w. 
2. There is a sqrt introduced somewhere between calculation and use.
3. My math is wrong.

You can modify the multifitproblem nllf method in bumps to print out the scaled nllf from the parts to see that they have approximately the same value.